File: src/main/java/br/com/blz/testjava/warehouse/WarehouseEntity.java
package br.com.blz.testjava.warehouse;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.sun.istack.NotNull;
import lombok.Data;

import javax.persistence.*;

@Data
@Entity
@Table(name = "warehouse")
public class WarehouseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "id_warehouse", unique = true, nullable = false)
    @JsonIgnore
    private Long id;

    @JoinColumn(name = "locality", nullable = false)
    @NotNull
    private String locality;

    @JoinColumn(name = "quantity", nullable = false)
    @NotNull
    private Integer quantity;

    @JoinColumn(name = "type", nullable = false)
    @NotNull
    private WarehouseType type;

}

File: src/main/java/br/com/blz/testjava/resource/repository/entity/WarehouseEntity.kt
package br.com.blz.testjava.resource.repository.entity

import br.com.blz.testjava.domain.entity.Warehouse
import br.com.blz.testjava.domain.entity.WarehouseType
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id

@Entity
class WarehouseEntity(
  @Id
  @GeneratedValue(strategy = GenerationType.AUTO)
  val id: Int = 0,

  val locality: String = "",
  val quantity: Int = 0,
  val type: String = "",
  val sku: Int = 0
) {
  fun toDomain() = Warehouse(
    locality = locality,
    quantity = quantity,
    type = WarehouseType.valueOf(type)
  )
  companion object {
    fun fromDomain(warehouse: Warehouse, sku: Int = 0) = WarehouseEntity(
      locality = warehouse.locality,
      quantity = warehouse.quantity,
      type = warehouse.type.name,
      sku = sku
    )
  }
}
